var/spack/repos/builtin/packages/cray-mpich/package.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class CrayMpich(Package):
+    """Cray's MPICH is a high performance and widely portable implementation of
+    the Message Passing Interface (MPI) standard."""
+
+    homepage = "https://docs.nersc.gov/development/compilers/wrappers/"
+    has_code = False    # Skip attempts to fetch source that is not available
+
+    maintainers = ['haampie']
+
+    version('7.7.16')
+    version('7.7.15')
+    version('7.7.14')
+    version('7.7.13')
+
+    provides('mpi@3:')
```

Comment:
`provides('mpi@3:')` might be asserting too much: that is saying this provides `mpi@4` (and all versions of MPI that could be released in the future). I think `provides('mpi@3')` is appropriate (and correct as mentioned in https://github.com/spack/spack/pull/20076#discussion_r543617512).

Reply:
Hm yeah, but it's also ambiguous imho... is it version "3" exactly or the range [3, 4). Or should it really represent version 3 exactly? I don't know how mpi is versioned.

Reply:
Spack doesn't have explicit exclusive range constraints, but specifying `@3` is equivalent to `[3, 4)`. MPI versions appear to be `x.y` (based on https://www.mpi-forum.org); I see several instances of `provides(mpi@:3.1)` (which means "this provides all MPI standards up to `3.1`"). Is there documentation on Cray MPICH which mentions the supported API versions?